addOns/replacer/src/main/java/org/zaproxy/zap/extension/replacer/ReplacerParam.java

```diff
@@ -275,4 +312,38 @@ public void setConfirmRemoveToken(boolean confirmRemove) {
         this.confirmRemoveToken = confirmRemove;
         getConfig().setProperty(CONFIRM_REMOVE_RULE_KEY, Boolean.valueOf(confirmRemoveToken));
     }
+
+    @Override
+    protected String getConfigVersionKey() {
+        return CONFIG_VERSION_KEY;
+    }
+
+    @Override
+    protected int getCurrentVersion() {
+        return CURRENT_CONFIG_VERSION;
+    }
+
+    @Override
+    @SuppressWarnings("fallthrough")
+    protected void updateConfigsImpl(int fileVersion) {
+        switch (fileVersion) {
+            case NO_CONFIG_VERSION:
+                // Handle unversioned to versioned update
+                parseReplacerRules();
```

Comment:
This leads to double parsing on update, but I guess it's fine.

Reply:
Ya that's the conclusion I came to as well. Not optimal but I couldn't see another way to do it.